Sources/Swift/Tools/UIRedactBuilder.swift

```diff
@@ -107,6 +82,24 @@ class UIRedactBuilder {
         redactClasses.forEach(addRedactClass(_:))
     }
     
+    /**
+     This function identifies and returns the regions within a given UIView that need to be redacted, based on the specified redaction options.
+     
+     - Parameter view: The root UIView for which redaction regions are to be calculated.
+     - Parameter options: A `SentryRedactOptions` object specifying whether to redact all text (`redactAllText`) or all images (`redactAllImages`). If `options` is nil, defaults are used (redacting all text and images).
+     
+     - Returns: An array of `RedactRegion` objects representing areas of the view (and its subviews) that require redaction, based on the current visibility, opacity, and content (text or images).
+     
+     The method recursively traverses the view hierarchy, collecting redaction areas from the view and all its subviews. Each redaction area is calculated based on the view’s presentation layer, size, transformation matrix, and other attributes.
+     
+     The redaction process considers several key factors:
+     1. **Text Redaction**: If `redactAllText` is set to true, regions containing text within the view or its subviews are marked for redaction.
+     2. **Image Redaction**: If `redactAllImages` is set to true, image-containing regions are also marked for redaction.
+     3. **Opaque View Handling**: If an opaque view covers the entire area, obfuscating views beneath it, those hidden views are excluded from processing, and we can remove them from the result.
+     4. **Clip Area Creation**: If a smaller opaque view blocks another view, we create a clip area to avoid drawing a redact mask on top of a view that does not require redaction.
+     
+     This function returns the redaction regions in reverse order from what was found in the view hierarchy, allowing the processing of regions from top to bottom. This ensures that clip regions are applied first before drawing a redact mask on lower views.
+     */
```

Comment:
I dont like step-by-step comment on the code when what it is doing is straight forward, the code should be its best documentation.

A high level explanation like this one to be used when you're calling the function I think its ok.

> I can't tell exactly what layerOriginalFrame

Returns the layer frame without transform (adding as a comment)

> Personally, I think this function should be rewritten to remove the inout

It would make it slower and much harder to understand the flow.